substrate/primitives/api/proc-macro/src/runtime_metadata.rs

```diff
@@ -298,14 +303,30 @@ pub fn generate_impl_runtime_metadata(impls: &[ItemImpl]) -> Result<TokenStream2
 	// Therefore, the `Deref` trait will resolve the `runtime_metadata` from `impl_runtime_apis!`
 	// when both macros are called; and will resolve an empty `runtime_metadata` when only the
 	// `construct_runtime!` is called.
-	Ok(quote!(
-		#crate_::frame_metadata_enabled! {
-			#[doc(hidden)]
-			impl #crate_::metadata_ir::InternalImplRuntimeApis for #runtime_name {
-				fn runtime_metadata(&self) -> #crate_::vec::Vec<#crate_::metadata_ir::RuntimeApiMetadataIR> {
-					#crate_::vec![ #( #metadata, )* ]
+	let block = match kind {
+		Kind::Main(paths) => {
+			quote! {
+				#[doc(hidden)]
+				impl #crate_::metadata_ir::InternalImplRuntimeApis for #runtime_name {
+					fn runtime_metadata(&self) -> #crate_::vec::Vec<#crate_::metadata_ir::RuntimeApiMetadataIR> {
+						let other_apis: #crate_::vec::Vec<#crate_::metadata_ir::RuntimeApiMetadataIR> = #crate_::vec::Vec::from([#crate_::vec![ #( #metadata, )*], #(#paths::runtime_metadata(),)*]).concat();
+						other_apis
+					}
 				}
 			}
+		},
+		Kind::Ext => quote! {
+					#[doc(hidden)]
+					#[inline(always)]
+					pub fn runtime_metadata() -> #crate_::vec::Vec<#crate_::metadata_ir::RuntimeApiMetadataIR> {
```

Comment:
I would name it `partial_runtime_metadata` to avoid confusion and suggestions from rustc